src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -290,22 +290,31 @@ private void WriteMvidsTo(Metadata metadata, ObjectWriter writer, CancellationTo
                 return;
             }
 
-            if (metadata is AssemblyMetadata assemblyMetadata)
+            try
             {
-                writer.WriteInt32((int)assemblyMetadata.Kind);
+                if (metadata is AssemblyMetadata assemblyMetadata)
+                {
+                    writer.WriteInt32((int)assemblyMetadata.Kind);
 
-                var modules = assemblyMetadata.GetModules();
-                writer.WriteInt32(modules.Length);
+                    var modules = assemblyMetadata.GetModules();
+                    writer.WriteInt32(modules.Length);
```

Comment:
:question: This method is no longer guaranteed to write the correct number of "MVIDs". What happens to deserialization for this case?